src/Microsoft.ComponentDetection.Detectors/nuget/NuGetProjectModelProjectCentricComponentDetector.cs

```diff
@@ -278,6 +278,13 @@ private void NavigateAndRegister(
         var libraryComponent = new DetectedComponent(new NuGetComponent(library.Name, library.Version.ToNormalizedString()));
         singleFileComponentRecorder.RegisterUsage(libraryComponent, explicitlyReferencedComponentIds.Contains(libraryComponent.Component.Id), parentComponentId);
 
+        // get the actual component in case it already exists
+        libraryComponent = singleFileComponentRecorder.GetComponent(libraryComponent.Component.Id);
+
+        // Add framework information to the actual component
+        var targetFramework = target.TargetFramework?.GetShortFolderName() ?? "unknown";
```

Comment:
If we don't know the framework (aka `unknown`, empty, or whitespace), let's not add anything to the `TargetFrameworks` set.

Reply:
This was intentional.  The addition of `TargetFrameworks` to the `NuGetComponent` is for diagnostics.  It's helpful to understand when a component was detected without a framework and be able to distinguish that from a case where a component was reported by a different detector (either old one, or `NuGetComponentDetector`)

Reply:
Actually it's never going to be the case here so I'm fine to just leave it out.  That's what I did for packages.config.